src/api/trackerTrackedEntities.ts

```diff
@@ -82,6 +82,7 @@ export interface AttributeValue {
 
 export interface Attribute {
     attribute: Id;
+    code?: string;
     valueType?: string;
     value: string;
 }
```

Comment:
I am not familiar with the responses of this endpoint, so I tried to gather data from play. After adding a code to attribute ASHR, I get:

```
$ curl -sS -u admin:district 'https://play.im.dhis2.org/stable-2-40-4-1/api/tracker/trackedEntities/FBVT1gwmdWi.json' | jq '.attributes'
[
  {
    "attribute": "zIu4WX5NfdD",
    "code: "ASRH",
    "displayName": "ASRH - Name of supervisor",
    "createdAt": "2024-07-17T15:57:00.475",
    "updatedAt": "2024-07-17T15:57:00.475",
    "storedBy": "admin",
    "valueType": "TEXT",
    "value": "samp"
  }
]
```

Now that we are on it, do we want to add all the other informative fields to the type? They will need to be optional, as code, as we reuse the same type for the GET (where we'll get a lot of informative fields) and for the POST (where only the compulsory attribute+value are required (right?), and certainly we won't want to make the other compulsory). 

For the future (or now, if we really have time for it), I think it's a good practice to have separate types for the GETs and for the POSTs, right?

Reply:
@tokland I  have added displayName, createdAt and updatedAt fields. However I have not added storedBy as it is not part of the response in our instance , could it be because of DHIS version difference? : 
```
https://metadata.eyeseetea.com/rsl-zebra/api/tracker/trackedEntities/cjEQBmguxfP.json
{
      "attribute": "baXJssWSL8r",
      "code": "RTSL_ZEB_TEA_DATE_NOTIFIED_NARRATIVE",
      "displayName": "Date Notified Narrative",
      "createdAt": "2024-07-23T13:39:47.148",
      "updatedAt": "2024-07-23T13:39:47.148",
      "valueType": "TEXT",
      "value": "notified"
    },
```

Is the idea to have a different Attribute for Get and Post so that the get one would have all the fields and the post will have only id and value because, we can only post/update value?

Reply:
Yes, as a rule of thumb, we should have different types for GET and POST, as POST types typically required less properties to be filled.